packages/phone-number-privacy/test/end-to-end/get-blinded-salt.test.ts

```diff
@@ -0,0 +1,108 @@
+import { PhoneNumberUtils } from '@celo/utils'
+import { normalizeAddressWith0x, privateKeyToAddress } from '@celo/utils/lib/address'
+import { serializeSignature, signMessage } from '@celo/utils/lib/signatureUtils'
+import 'isomorphic-fetch'
+import { getBlindedPhoneNumber } from '../utils'
+
+require('dotenv').config()
+
+const PHONE_NUM_PRIVACY_SERVICE = process.env.PHONE_NUM_PRIVACY_SERVICE_URL
+const SIGN_MESSAGE_ENDPOINT = '/getBlindedSalt'
+
+const PRIVATE_KEY1 = '0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef'
+const ACCOUNT_ADDRESS1 = normalizeAddressWith0x(privateKeyToAddress(PRIVATE_KEY1))
+const PRIVATE_KEY2 = '0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890fdeccc'
+const ACCOUNT_ADDRESS2 = privateKeyToAddress(PRIVATE_KEY2)
+const PHONE_NUMBER = '+15555555555'
+const IDENTIFIER = PhoneNumberUtils.getPhoneHash(PHONE_NUMBER)
+const BLINDING_FACTOR = new Buffer('0IsBvRfkBrkKCIW6HV0/T1zrzjQSe8wRyU3PKojCnww=', 'base64')
+const BLINDED_PHONE_NUMBER = getBlindedPhoneNumber(PHONE_NUMBER, BLINDING_FACTOR)
+
+describe('Running against a deployed service', () => {
+  describe('Returns status 400 with invalid input', () => {
+    it('With invalid address', async () => {
+      const response = await postToSignMessage(
+        BLINDED_PHONE_NUMBER,
+        PRIVATE_KEY1,
+        '0x1234',
+        'ignore'
+      )
+      expect(response.status).toBe(400)
+    })
+
+    it('With missing blindedQueryPhoneNumber', async () => {
+      const response = await postToSignMessage('', PRIVATE_KEY1, ACCOUNT_ADDRESS1)
+      expect(response.status).toBe(400)
+    })
+  })
+
+  describe('Returns status 401 with invalid authentication headers', () => {
+    it('With invalid auth header', async () => {
+      const response = await postToSignMessage(
+        BLINDED_PHONE_NUMBER,
+        PRIVATE_KEY1,
+        ACCOUNT_ADDRESS1,
+        'invalid'
+      )
+      expect(response.status).toBe(401)
+    })
+
+    it('With auth header signer mismatch', async () => {
```

Comment:
might also be nice to have test with happy path -- correct auth header

Reply:
Agreed! I wanted to add this but the account will hit the quota limit unless it's prefunded (hence the out of quota limit test is really the auth happy path). I added a new, disabled test to test the 200 case but it requires the tester to prefund the test account to get past the quota limit.